generatorreceiver/internal/flags/flag_embed.go

```diff
@@ -10,7 +10,7 @@ type EmbeddedFlags struct {
 	FlagUnset string `json:"flag_unset" yaml:"flag_unset"`
 }
 
-func (f *EmbeddedFlags) ShouldGenerate() bool {
+func (f EmbeddedFlags) ShouldGenerate() bool {
```

Comment:
I changed these pointer receivers to non pointer receivers because it complicated implementing the generic function. I can elaborate if needed, but I figured this was a safe change because we never modify the flags in these functions, we only read them.